x-pack/plugins/observability/public/pages/alerts/index.tsx

```diff
@@ -103,36 +102,62 @@ function AlertsPage() {
     ];
   }, [indexNames]);
 
-  const setWorkflowStatusFilter = useCallback(
-    (value: AlertWorkflowStatus) => {
-      setWorkflowStatus(value);
-    },
-    [setWorkflowStatus]
-  );
+  // Keep the Workflow status code commented (no delete) as requested: https://github.com/elastic/kibana/issues/117686
+
+  // const setWorkflowStatusFilter = useCallback(
+  //   (value: AlertWorkflowStatus) => {
+  //     setWorkflowStatus(value);
+  //   },
+  //   [setWorkflowStatus]
+  // );
 
   const onQueryChange = useCallback(
     ({ dateRange, query }) => {
       if (rangeFrom === dateRange.from && rangeTo === dateRange.to && kuery === (query ?? '')) {
         return refetch.current && refetch.current();
       }
-
       timefilterService.setTime(dateRange);
       setRangeFrom(dateRange.from);
       setRangeTo(dateRange.to);
       setKuery(query);
+      syncAlertStatusFilterStatus(query as string);
     },
     [rangeFrom, setRangeFrom, rangeTo, setRangeTo, kuery, setKuery, timefilterService]
   );
 
-  const addToQuery = useCallback(
-    (value: string) => {
-      let output = value;
-      if (kuery !== '') {
-        output = `${kuery} and ${value}`;
+  const syncAlertStatusFilterStatus = (query: string) => {
+    const [, alertStatus] = /\s*kibana\.alert\.status\s*:\s*"(.*?)"/.exec(query) || [];
```

Comment:
@fkanout We shouldn't be using the hardcoded `kibana.alert.status` value here

Reply:
I understand your point. However, it's a RegEx, so that I will construct it from the `technical_field_names`. I'm not sure if the implementation will look good 🤔

Reply:
Done